### PR TITLE
Added Suite of Portfolio Functions

### LINKED ISSUE #23: Enhancement: Connect a Stock Portfolio API and track user investments
This will likely use yfinance for retrieving the stock price data.

### LINKED ISSUE #46: Bug in Expense Tracking: Fails to track expense if no users have tracked expenses before
This has been fixed by adjusting the add_user_record method in add.py. This fix is in my feature/portfolio branch

### PR DESCRIPTION
**Pull request by Nlorenc2760**
1. Buy Stocks at current prices
2. Sell Stocks at current prices
3. View Current Portfolio and percent changes

Close #23 
Close #46

### CODE REVIEW

File: code/portfolio_buy.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    # Check to see if the stock is real, and get its price if it is
+    stock = yf.Ticker(ticker)
+    if stock is not None:
+        msg = bot.send_message(chat_id, "Enter Number of Shares")
+        bot.register_next_step_handler(msg, handle_stock_buy, bot, ticker, stock)
+    else:
+        bot.send_message(chat_id, "Invalid Stock")
+
+def handle_stock_buy(message, bot, ticker, stock):
+    try:
```
Comment (Kethly) [inline code/portfolio_buy.py:35]: Add a function comment here

File: code/portfolio_sell.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    which is the telegram bot object from the main code.py function. It gets the user's chat ID
+    from the message object, and reads all user data through the read_json method from the helper module.
+    It then proceeds to empty the budget data for the particular user based on the user ID provided from the UI.
+    It returns a simple message indicating that this operation has been done to the UI.
+    """
+    chat_id = message.chat.id
+    msg = bot.send_message(chat_id, "Enter a Stock Ticker")
+    bot.register_next_step_handler(msg, handle_stock_name, bot)
+
+def handle_stock_name(message, bot):
```
Comment (Kethly) [inline code/portfolio_sell.py:22]: here too

File: code/portfolio_sell.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    # Check to see if the stock is real, and get its price if it is
+    stock = yf.Ticker(ticker)
+    if stock is not None:
+        msg = bot.send_message(chat_id, "Enter Number of Shares")
+        bot.register_next_step_handler(msg, handle_stock_sell, bot, ticker, stock)
+    else:
+        bot.send_message(chat_id, "Invalid Stock")
+
+def handle_stock_sell(message, bot, ticker, stock):
```
Comment (Kethly) [inline code/portfolio_sell.py:34]: here as well

File: code/portfolio_view.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    from the message object, and reads all user data through the read_json method from the helper module.
+    It then proceeds to empty the budget data for the particular user based on the user ID provided from the UI.
+    It returns a simple message indicating that this operation has been done to the UI.
+    """
+    chat_id = message.chat.id
+    bot.send_message(chat_id, "Here is your Portfolio:")
+    viewPortfolioTable(message, bot)
+    
+
+def viewPortfolioTable(message, bot):
```
Comment (Kethly) [inline code/portfolio_view.py:20]: One comment here

**Review by Kethly — CHANGES_REQUESTED** [review]

File: code/portfolio_buy.py (hunk 1 above)
Comment (Nlorenc2760) [reply code/portfolio_buy.py:41]: A comment has now been added.

File: code/portfolio_sell.py (hunk 2 above)
Comment (Nlorenc2760) [reply code/portfolio_sell.py:22]: A comment has now been added.

File: code/portfolio_sell.py (hunk 3 above)
Comment (Nlorenc2760) [reply code/portfolio_sell.py:37]: A comment has now been added.

File: code/portfolio_view.py (hunk 4 above)
Comment (Nlorenc2760) [reply code/portfolio_view.py:20]: A comment has now been added.

**Comment by Kethly** [comment]
Looks great.

**Pull request merged**